docs/drafts/version-ranges.md

```diff
@@ -0,0 +1,49 @@
+# Operator version ranges
+
+This document explains how to specify a version range to install or update an Operator with OLM 1.0.
+
+You define an Operator's target version in its custom resource (CR) file.
+The following list describes how OLM resolves an Operator's target version, and the resulting actions:
+
+Not specifying a version in the CR
+: Installs or updates the latest version of the Operator.
```

Comment:
> latest

Technically, it's the latest version _that satisfies resolution_. I think it's important to call that out pretty clearly somewhere, and possibly explain with an example.

For example:

In the catalog:
- packageA v1.0.0 exists and depends on packageB v1.2.0
- packageB v1.2.0 exists
- packageB v1.2.1 exists

On the cluster:
- Operator for `packageA` says `spec.version: 1.0.0`
- Operator for `packageB` says `spec.version: 1.2.x`

The version that will ultimately be chosen for `packageB` is 1.2.0 even though 1.2.1 is available, because that's what `packageA` requires.

Reply:
For the moment, I linked to Deppy's README to explain resolution. WDYT of creating a follow up issue to port the OCP docs about dependency resolution upstream?